src/Admin/AjaxAction.php

```diff
@@ -0,0 +1,225 @@
+<?php
+/**
+ * Class AjaxAction.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\Admin;
+
+/**
+ * Base class to define a new AJAX action.
+ *
+ * @package AmpProject\AmpWP
+ */
+final class AjaxAction {
+
+
+	/**
+	 * The AJAX action should be accessible by any visitor.
+	 *
+	 * @var int
+	 */
+	const ACCESS_ANY = 0;
+
+	/**
+	 * The AJAX action should only be accessible by authenticated users.
+	 *
+	 * @var int
+	 */
+	const ACCESS_USER = 1;
+
+	/**
+	 * The AJAX action should only be accessible by unauthenticated visitors.
+	 *
+	 * @var int
+	 */
+	const ACCESS_NO_USER = 2;
+
+	/**
+	 * The AJAX handler listens on both the backend and the frontend.
+	 *
+	 * @var int
+	 */
+	const SCOPE_BOTH = 0;
+
+	/**
+	 * The AJAX handler listens on the backend only.
+	 *
+	 * @var int
+	 */
+	const SCOPE_BACKEND = 1;
+
+	/**
+	 * The AJAX handler listens on the frontend only.
+	 *
+	 * @var int
+	 */
+	const SCOPE_FRONTEND = 2;
+
+	/**
+	 * Action to use for enqueueing the JS logic at the frontend.
+	 *
+	 * @var string
+	 */
+	const FRONTEND_ENQUEUE_ACTION = 'wp_enqueue_scripts';
+
+	/**
+	 * Action to use for enqueueing the JS logic at the backend.
+	 *
+	 * @var string
+	 */
+	const BACKEND_ENQUEUE_ACTION = 'admin_enqueue_scripts';
```

Comment:
I think this needs to be more fine-grained. We only want it to fire on the site health screen, not on any other admin page.

Reply:
Right now, this only starts the check to see if we're on the Site Health screen. The selector to attach the click event to is the very first thing that is being checked with an early return, so nothing happens when our button is not present.

Reply:
Something is still happening, though: CPU cycles to parse and evaluate the JS. This JS is being enqueued on the page which may not need it, namely `wp-util`. None of this JS should appear on any page other than Site Health.

Reply:
No, it will only ever on the page that contains this "button": `a.reenable-css-transient-caching`. That is the very first early return in the method that is hook to this action:

https://github.com/ampproject/amp-wp/pull/4522/files#diff-e97af8c0e681095ef90cd07c4cea405dR166-R168

As that button will only be present on the Site Health page, the JS will also only be enqueued there.

Reply:
I wrote nonsense above, as the code is only checking whether the variable is set, not whether a query against the DOM succeeded.

I'm adapting the code to use the `$hook_suffix` argument for the `admin_enqueue_scripts` action.